mne/preprocessing/stim.py

```diff
@@ -67,3 +91,92 @@ def eliminate_stim_artifact(raw, events, event_id, tmin=-0.005,
         elif mode == 'window':
             raw[picks, first_samp:last_samp] = data * window[np.newaxis, :]
     return raw
+
+
+def fix_stim_artifact(inst, events=None, event_id=None, tmin=0.,
+                      tmax=0.01, mode='linear', copy=False):
+    """Eliminates stimulations artifacts from instance
+
+    The instance will be modified in place (no copy)
+
+    Parameters
+    ----------
+    inst : instance of evoked or epochs
+        instance
+    events : array, shape (n_events, 3)
+        The list of events. No need when inst is epochs or evoked
+    event_id : int
+        The id of the events generating the stimulation artifacts.
+        No need when inst is epochs or evoked
+    tmin : float
+        Start time of the interpolation window in seconds.
+    tmax : float
+        End time of the interpolation window in seconds.
+    mode : 'linear' | 'window'
+        way to fill the artifacted time interval.
+        'linear' does linear interpolation
+        'window' applies a (1 - hanning) window.
+    copy : bool
+        If False inst is cropped in place.
+
+    Returns
+    -------
+    inst : instance of evoked or epochs
+        instance with modified data
+    """
+    if copy:
+        inst = inst.copy()
+    s_start = int(np.ceil(inst.info['sfreq'] * tmin))
+    s_end = int(np.ceil(inst.info['sfreq'] * tmax))
+    if (s_end - s_start - 4) < 0:
+        raise ValueError('Time between tmin and tmax is too short'
+                         ' to fix artifact. Input longer values')
+    window = None
+    if mode == 'window':
+        window = _get_window(s_start, s_end)
+    ch_names = inst.info['ch_names']
+    picks = pick_channels(ch_names, ch_names)
+
+    if isinstance(inst, Raw):
+        if inst.preload is False:
+            raise RuntimeError('Modifying data of Raw is only supported '
+                               'when preloading is used. Use preload=True '
+                               '(or string) in the constructor.')
+        events_sel = (events[:, 2] == event_id)
+        event_start = events[events_sel, 0]
+        if mode is 'linear':
+            data, _ = inst[picks, :]
+        elif mode is 'window':
+            data, _ = inst[:, :]
+        for k in range(len(event_start)):
+            first_samp = int(event_start[k]) - inst.first_samp + s_start
+            last_samp = int(event_start[k]) - inst.first_samp + s_end
+            _fix_artifact(inst, data, window, picks, first_samp, last_samp, mode)
+
+    elif isinstance(inst, Epochs):
+        if inst.preload is False:
+            raise RuntimeError('Modifying data of Epochs is only supported '
```

Comment:
@mainakjas I think preload check is needed. When Epochs is not preloaded, get_data method return data from _get_data_from_disk(). Whenever I call methed get_data, it returns data from  _get_data_from_disk() method which re-load data from raw. So it will not affect epochs' data.  But when Epochs is preloaded, get_data() method will return epcohs._data. By modifying it, fix_stim_artifact will change real data epoch have.